src/main/java/jenkins/scm/api/SCMFileSystem.java

```diff
@@ -158,6 +158,75 @@ public boolean changesSince(@CheckForNull SCMRevision revision, @NonNull OutputS
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Given a {@link SCM} this method will try to retrieve a corresponding {@link SCMFileSystem} instance.
+     *
+     * @param build the build of the {@link SCM}
+     * @param scm the {@link SCM}.
+     * @return the corresponding {@link SCMFileSystem} or {@code null} if there is none.
+     * @throws IOException          if the attempt to create a {@link SCMFileSystem} failed due to an IO error
+     *                              (such as the remote system being unavailable)
+     * @throws InterruptedException if the attempt to create a {@link SCMFileSystem} was interrupted.
+     */
+    @CheckForNull
+    public static SCMFileSystem of(@NonNull Run build, @NonNull SCM scm) throws IOException, InterruptedException {
+        return of(build, scm, null);
+    }
+
+    /**
+     * Given a {@link SCM} this method will try to retrieve a corresponding {@link SCMFileSystem} instance that
+     * reflects the content at the specified {@link SCMRevision}.
+     *
+     * @param build the build of the {@link SCM}
+     * @param scm the {@link SCM}.
+     * @param rev the specified {@link SCMRevision}.
+     * @return the corresponding {@link SCMFileSystem} or {@code null} if there is none.
+     * @throws IOException          if the attempt to create a {@link SCMFileSystem} failed due to an IO error
+     *                              (such as the remote system being unavailable)
+     * @throws InterruptedException if the attempt to create a {@link SCMFileSystem} was interrupted.
+     */
+    @CheckForNull
+    public static SCMFileSystem of(@NonNull Run build, @NonNull SCM scm, @CheckForNull SCMRevision rev)
```

Comment:
Ah, so there's an `Item owner` version of this and `Run build` version.

Reply:
I tried to think of a good way to try and deduplicate this code without much success as both `supports` and `builder.build` has multiple implementations.

Yes `Run` does implement `Item` however I am unsure how to the write method to deduplicate.
Also `supports` has several more implementations and `of` also includes `SCMSource.class, SCMHead.class`